virttest/graphical_console.py

```diff
@@ -31,6 +34,45 @@ def get_keymap(keymap_file):
         return keymap_dict
 
 
+def uniform_linear(duration, rate):
+    """
+    Uniform linear trace for mouse move.
+
+    :param duration: the move duration time.
+    :param rate: frequency of sending move event.
+    :param src: start position.
+    :param dst: destination position.
+
+    :return: the object with a position in this line.
+    :rtype: function for motion.
+    """
+
+    def motion(src, dst):
+        x1, y1 = src
+        x2, y2 = dst
+        x, y = x1, y1
+        num = duration * rate
+        dx_int, dx_float = str((x2 - x1) / num).split('.')
+        dy_int, dy_float = str((y2 - y1) / num).split('.')
+        decimal_x = 0
+        decimal_y = 0
+        for i in range(num):
+            decimal_x += float("0." + dx_float)
+            decimal_y += float("0." + dy_float)
+            if decimal_x >= 1:
+                x += (-1 if "-" in str(dx_int) else 1)
+                decimal_x = float("0." + (str(decimal_x).split(".")[1]))
+            if decimal_y >= 1:
+                y += (-1 if "-" in str(dy_int) else 1)
+                decimal_y = float("0." + (str(decimal_y).split(".")[1]))
+            x, y = x + int(dx_int), y + int(dy_int)
+            if i == num - 1:
+                x, y = x2, y2
+            yield (x, y)
+            time.sleep(duration / rate)
```

Comment:
not correct, please reference the above comment.

Reply:
updated~